src/hosting-node.js

```diff
@@ -51,64 +52,88 @@ async function getBody(url, userAgentIdentifier) {
 /**
  * Check if a domain is hosted by a green web host.
  * @param {string|array} domain - The domain to check, or an array of domains to be checked.
- * @param {object} db - Optional. A database object to use for lookups.
- * @param {string} userAgentIdentifier - Optional. The app, site, or organisation that is making the request.
- * @returns {boolean|array} - A boolean if a string was provided, or an array of booleans if an array of domains was provided.
+ * @param {string[] | DomainCheckOptions} optionsOrDb - Optional. An object of domain check options, or a database list to use for lookups.
+ * @param {string | DomainCheckOptions} userAgentIdentifier - Optional. The app, site, or organisation that is making the request.
```

Comment:
minor: now that this param can only be the `userAgentIdentifier`, change the type back to `{string}`